### PR TITLE
PR to return peakrange in initial processing of spin echo data, need …

### PR DESCRIPTION
**Pull request by wfkincaid**
…to fix set_prop() still

### CODE REVIEW

File: pyspecProcScripts/phasing.py [hunk 1]
```diff
@@ -320,6 +321,8 @@ def filter_ranges(B, A):
     peakrange = peakrange[0]
     frq_center = np.mean(peakrange).item()
     frq_half = np.diff(peakrange).item() / 2
+    d.set_prop(peakrange, frq_half)
```
Comment (jmfranck) [inline pyspecProcScripts/phasing.py:324]: The first argument should be a string. This basically gets loaded into a dictionary.

Comment (jmfranck) [reply pyspecProcScripts/phasing.py:324]: In other words, the first argument is like a key

File: pyspecProcScripts/phasing.py [hunk 2]
```diff
@@ -320,6 +321,8 @@ def filter_ranges(B, A):
...
+    print(peakrange)
```
Comment (jmfranck) [inline pyspecProcScripts/phasing.py:325]: This is fine for debugging, but don't include it in the final version that you want to merge

**Review by jmfranck — CHANGES_REQUESTED** [review]

File: pyspecProcScripts/phasing.py [hunk 3]
```diff
@@ -468,6 +469,7 @@ def filter_ranges(B, A):
     d[direct, 0] *= 0.5
     d.ft(direct)
     return d
+    return peakrange
```
Comment (jmfranck) [inline pyspecProcScripts/phasing.py:472]: just remove this line.  It's actually not doing anything, anyways.  And, given that you set the `"peakrange"` property above, it's now accessible as `d.get_prop("peakrange")`

Comment (wfkincaid) [reply pyspecProcScripts/phasing.py]: removed

File: pyspecProcScripts/phasing.py [hunk 4]
```diff
@@ -753,7 +754,7 @@ def hermitian_function_test(
         return echo_peak + min_echo
 
 
-def determine_sign(s, direct="t2", fl=None):
+def determine_sign(s, direct="t2", basename=None, fl=None):
```
Comment (jmfranck) [inline pyspecProcScripts/phasing.py:757]: Rather, below replace the "if basename" with `if not hasattr(fl,"basename") or fl.basename is None`

**Review by jmfranck — CHANGES_REQUESTED** [review]

**Comment by jmfranck** [comment]
merging in master, some things changed around.  Please review and check that it works the way you want it to.

**Comment by jmfranck** [comment]
(it's actually best to provide a very short example that demonstrates *only* the feature you're interested in, if you can do that in 30min)

File: pyspecProcScripts/phasing.py [hunk 5]
```diff
@@ -283,14 +285,20 @@ def fid_from_echo(
             alpha=0.25,
             label="half width",
         )
+        axvline(
+            x=frq_center + frq_half,
+            color="k",
+            ls=":",
+            alpha=0.25,
+            label="half width",
+        )
```
Comment (wfkincaid) [inline pyspecProcScripts/phasing.py:294]: I did this to match the other side, wasn't labeled previously

File: pyspecProcScripts/phasing.py [hunk 6]
```diff
@@ -136,11 +141,10 @@ def ph1_real_Abs(s, dw, ph1_sel=0, ph2_sel=1, fl=None):
...
 
     .. todo::
-        update with `sphinxcontrib-bibtex <https://sphinxcontrib-bibtex.readthedocs.io/en/latest/usage.html>`_.
+        update with `sphinxcontrib-bibtex
+        <https://sphinxcontrib-bibtex.readthedocs.io/en/latest/usage.html>`_.
 
-    Parameters
-    ==========
-    s: nddata
+    Parameters ========== s: nddata
```
Comment (jmfranck) [inline pyspecProcScripts/phasing.py:147]: This got messed up

File: pyspecProcScripts/phasing.py [hunk 7]
```diff
@@ -371,15 +381,16 @@ def fid_from_echo(
             if (
                 s_flipped.getaxis(direct)[idx] == 0
                 and d_sigcoh.getaxis(direct)[idx] == 0
-            ):  # should be centered about zero, but will not be if too lopsided
-                for_resid = (
+            ):  # should be centered about zero,
```
Comment (jmfranck) [inline pyspecProcScripts/phasing.py:384]: Indentation of comment weird

File: pyspecProcScripts/phasing.py [hunk 8]
```diff
@@ -371,15 +381,16 @@ def fid_from_echo(
...
+                for_resid = ( 
                     abs(s_flipped - d_sigcoh[direct:(t_start, -t_start)]) ** 2
                 )
                 N_ratio = for_resid.data.size
                 for_resid.mean_all_but(direct).run(sqrt)
-                N_ratio /= (
-                    for_resid.data.size
-                )  # the signal this has been plotted against is signal averaged by N_ratio
+                N_ratio /= (for_resid.data.size) 
+                # the signal this has been plotted against is signal averaged
```
Comment (jmfranck) [inline pyspecProcScripts/phasing.py:392]: Comment should be at end of line

File: pyspecProcScripts/phasing.py [hunk 9]
```diff
@@ -422,12 +433,11 @@ def fid_from_echo(
...
 def find_peakrange(d, direct="t2", peak_lower_thresh=0.1, fl=None):
-    """find the range of frequencies over which the signal occurs, so that we can autoslice
+    """find the range of frequencies over which the signal occurs, so that we
+    can autoslice
 
-    Parameters
-    ==========
-    d : nddata
-        Data in the frequency domain -- will not be altered.
+    Parameters ========== d : nddata Data in the frequency domain -- will not
```
Comment (jmfranck) [inline pyspecProcScripts/phasing.py:439]: Got messed up

**Review by jmfranck — CHANGES_REQUESTED** [review]

**Comment by jmfranck** [comment]
thanks!

**Pull request merged**